src/fee_lab_to_nwb/scherrer_ophys/audiointerface.py

```diff
@@ -0,0 +1,66 @@
+from typing import Optional
+
+from neuroconv.basedatainterface import BaseDataInterface
+from neuroconv.utils import get_schema_from_hdmf_class, get_base_schema
+from pynwb import NWBFile, TimeSeries
+from scipy.io import wavfile
+
+from ndx_sound import AcousticWaveformSeries
+
+
+class AudioInterface(BaseDataInterface):
+    """Data interface for writing acoustic recording to an NWB file."""
+
+    def __init__(self, file_path: str):
+        """
+        Create the interface for writing acoustic recording to an NWB file.
+
+        Parameters
+        ----------
+        file_path: str
+            The path to the audio file.
+        """
+        super().__init__(file_path=file_path)
+
+    def get_metadata_schema(self):
+        metadata_schema = super().get_metadata_schema()
+        time_series_metadata_schema = get_schema_from_hdmf_class(TimeSeries)
+        exclude = ["conversion", "starting_time", "rate"]
+        for key in exclude:
+            time_series_metadata_schema["properties"].pop(key)
+        metadata_schema["properties"]["Behavior"] = get_base_schema(tag="Behavior")
+        metadata_schema["properties"]["Behavior"].update(
+            required=["Audio"],
+            properties=dict(
+                Audio=dict(
+                    type="array",
+                    minItems=1,
+                    items=time_series_metadata_schema,
+                )
+            ),
+        )
+        return metadata_schema
+
+    def run_conversion(
+        self,
+        nwbfile: Optional[NWBFile] = None,
+        metadata: Optional[dict] = None,
+    ):
+
+        audio_metadata = metadata["Behavior"]["Audio"][0]
+
+        # Load the audio file.
+        file_path = self.source_data["file_path"]
+        sampling_rate, samples = wavfile.read(file_path, mmap=True)
+
+        # create AcousticWaveformSeries with ndx-sound
+        acoustic_waveform_series = AcousticWaveformSeries(
+            name=audio_metadata["name"],
+            data=samples,  # TODO: wrap into H5DataIO?
```

Comment:
the file that i was testing `micData_210729.wav` is 400Mb

Reply:
That's with `.wav` compression though I assume? How big is the array you read in and what `dtype` is it?

Reply:
dtype is `int16`, ndarray size 417.81 MB reading with `wavfile.read(file_path, mmap=False)`

Reply:
Does `mmap=True` work at all and if so is it compatible with a DataChunkIterator (old one or a custom new Generic-based one)?

Just trying to think about future-proofing this script for even longer recordings, since this is almost half of a full buffer as is

Reply:
This nwb file was written with mmap=True, and I had no problem with reading it. I'm not sure how compatible it is with the iterators though, I can def try it!

Reply:
Great, then if so I'd keep it set to `True` always

Reply:
Are we likely to get wav files that have trouble fitting in memory? This would be a good idea eventually but I doubt it's on the critical path for this project. How big is the hour-long wav file? I do think applying compression will probably help a lot though

Reply:
If they double the length of the current session it will be close to the default buffer size.

If they ever do really long recordings (8+ hours like we've seen from other labs) then it would actually become a concern.

My point though is it will take us only a couple of minutes to implement it.

Also if we use the `GenericDataChunkIterator` it will set better default chunk sizes than `h5py` does.

Actually, you should be able to just use the `SliceableDataChunkIterator` from `neuroconv` as-is without any extra work, right? @weiglszonja

Reply:
yes it works just fine.

Reply:
Great!